### PR TITLE
Add roadmap for O11y and Edge

### PR DESCRIPTION
**Pull request by benjaminhuo**
Signed-off-by: Benjamin <benjamin@yunify.com>

**What type of PR is this?**
/kind documentation

### CODE REVIEW

**Review by wanjunlei — APPROVED** [review]

**Comment by benjaminhuo** [comment]
@wanjunlei @junotx Pls remember to submit PR to ks-installer if necessary for upgrade to fluenbit operator, notification manager etc.

File: docs/roadmap.md [hunk 1]
```diff
@@ -46,6 +46,41 @@ KubeSphere Roadmap demonstrates a list of open source product development plans
 - [x] Interactive creation pipeline. [#1283](https://github.com/kubesphere/console/issues/1283)
 - [x] Add S2I webhook support. [#6](https://github.com/kubesphere/s2ioperator/issues/6)
 
+### KubeEdge Integration [#3070](https://github.com/kubesphere/kubesphere/issues/3070)
+
+- [ ] KubeEdge cloud components setup.
+- [ ] KubeEdge edge nodes setup.
+- [x] Edge nodes logging and metrics support.
+- [x] Automate edge node joining/leaving.
```
Comment (Ma-Dan) [inline docs/roadmap.md:54]: "Automated edge node network configuration on joining/leaving" seems more clear?

Comment (benjaminhuo) [reply docs/roadmap.md:54]: Automated network configuration on edge node joining/leaving ?

Comment (Ma-Dan) [reply docs/roadmap.md:54]: Yes, this represents the function of edge-watcher with iptables-operator. Automate edge node joining is what we planned to do for more simplified joining process than keadm.

Comment (benjaminhuo) [reply docs/roadmap.md:54]: I think we still missing one item which is `Scheduling workload to edge nodes`

Comment (Ma-Dan) [reply docs/roadmap.md]: And one detail should also be fixed, automatic taint edge node on joining, not done yet. And workload of cloud, such as daemonsets should not be scheduled to edge nodes.

Comment (benjaminhuo) [reply docs/roadmap.md]: @Ma-Dan pls review again and approve the pr with lgtm if no problem

**Review by Ma-Dan — APPROVED** [review]

**Comment by benjaminhuo** [comment]
@Ma-Dan   / lgtm label is needed

**Review by Ma-Dan — APPROVED** [review]
/lgtm

**Pull request merged**